src/pretix/base/services/orders.py

```diff
@@ -1015,6 +1015,16 @@ def _recalculate_total_and_payment_fee(self):
         self.order.total += sum([f.value for f in self.order.fees.all()])
         self.order.save()
 
+    def _payment_fee_diff(self):
+        prov = self._get_payment_provider()
+        if prov:
+            old_total = sum([p.price for p in self.order.positions.all()])
+            new_total = old_total + self._totaldiff
+            old_fee = self.order.fees.get(fee_type=OrderFee.FEE_TYPE_PAYMENT).value
```

Comment:
We need to get the fee from the model, as some settings could have changed with the payment provider or the fees could depend on some economical factor.